### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by kaendfinger**
See the commits for more details, but this fixes quite a few bugs.

### CODE REVIEW

**Comment by kaendfinger** [comment]
cc @sethladd @keertip

File: lib/src/model.dart [hunk 1]
```diff
@@ -325,9 +325,12 @@ abstract class ModelElement {
           buf.write(p.modelType.element.linkedParams);
           buf.write(')');
         } else if (p.modelType != null &&
-            p.modelType.element != null &&
-            p.modelType.element.modelType != null) {
-          String typeName = p.modelType.element.modelType.linkedName;
+            p.modelType.element != null) {
```
Comment (keertip) [inline lib/src/model.dart]: Could you add a test for this fix in `test/fake-package` and `test/model_test.dart`?

Comment (kaendfinger) [inline lib/src/model.dart]: Done

**Comment by kaendfinger** [comment]
@keertip Ok, it's ready now :)

**Pull request merged**